ui/src/components/ResultBlock/index.tsx

```diff
@@ -15,8 +19,106 @@ const ResultBlockWrapper = styled.div`
   }
 `;
 
-export const ResultBlock: React.FC<{ result: any }> = ({ result }) => (
-  <ResultBlockWrapper>
-    {result.failureMode && result.failureMessage}
-  </ResultBlockWrapper>
-);
+const ResultSectionEl = styled.div`
+  margin-bottom: 1.5rem;
+`;
+
+const ResultLineEl = styled.div`
+  margin: 0.5rem 0;
+`;
+
+const CollapseBtnEl = styled.div`
+  margin: 1rem 0;
+`;
+
+const CollapsibleLogEl = styled.section`
+  max-height: ${({ isCollapsed }: { isCollapsed: boolean }) =>
+    isCollapsed ? "0rem" : "40rem"};
+  overflow: hidden;
+  transition: all 0.25s ease-in-out;
+`;
+
+interface NetworkCall {
+  request: Request & { headers: { [key: string]: string[] } };
+  response: Response;
+}
+
+export const ResultBlock: React.FC<{ result: any }> = ({ result }) => {
+  const [isCollapsed, setIsCollapsed] = useState(true);
+  return (
+    <ResultBlockWrapperEl>
+      <ResultSectionEl>
+        <Eyebrow>Failure Type:</Eyebrow>
+        {result.failureMode}
+      </ResultSectionEl>
+      <ResultSectionEl>
+        <Eyebrow>Description:</Eyebrow>
+        {result.failureMessage}
+      </ResultSectionEl>
+      <ResultSectionEl>
+        <ResultLineEl>
+          Expected: <strong>{result.expected}</strong>
+        </ResultLineEl>
+        <ResultLineEl>
+          Received: <strong>{result.actual}</strong>
+        </ResultLineEl>
+      </ResultSectionEl>
+      <CollapseBtnEl>
+        <TextLink onClick={() => setIsCollapsed(!isCollapsed)}>
+          {isCollapsed ? "Show" : "Collapse"} full error
+        </TextLink>
+      </CollapseBtnEl>
+
+      <CollapsibleLogEl isCollapsed={isCollapsed}>
+        <Log>
+          Network Calls:
+          {result.networkCalls.map((networkCall: NetworkCall, i: number) => (
+            <LogIndent key={`${networkCall.request.method}-${i}`}>
+              {networkCall.request && (
+                <>
+                  <div>Request:</div>
+                  <LogIndent>
+                    {networkCall.request.method} {networkCall.request.url}
+                    <div>Headers:</div>
+                    <LogIndent>
+                      content-type:{" "}
+                      {(
+                        networkCall.request?.headers?.["Content-Type"] || []
+                      ).map((contentType: string) => (
+                        <span key={contentType}>contentType</span>
+                      ))}
+                    </LogIndent>
+                  </LogIndent>
+
+                  <div>Body:</div>
+                  <div>{JSON.stringify(networkCall.request.body)}</div>
```

Comment:
I think we can use the `Json` component I added for the config modal here. We can't test it with testanchor.stellar.org since there are no errors that return JSON, but I think this will print out the JSON string with no formatting.

Reply:
Let's merge your PR and I'll add that here